scripts/linters/pre_commit_linter.py

```diff
@@ -0,0 +1,542 @@
+            verbose_mode_enabled=verbose_mode_enabled)
+        third_party_linters.append(third_party_linter)
+
+    if py_file_extension_type:
+        custom_linter, third_party_linter = python_linter.get_linters(
+            py_filepaths, verbose_mode_enabled=verbose_mode_enabled)
+        custom_linters.append(custom_linter)
+        third_party_linters.append(third_party_linter)
+
+    return custom_linters, third_party_linters
+
+
+def _get_changed_filepaths():
+    """Returns a list of modified files (both staged and unstaged)
+
+    Returns:
+        a list of filepaths of modified files.
+    """
+    unstaged_files = subprocess.check_output([
+        'git', 'diff', '--name-only',
+        '--diff-filter=ACM']).splitlines()
+    staged_files = subprocess.check_output([
+        'git', 'diff', '--cached', '--name-only',
+        '--diff-filter=ACM']).splitlines()
+    all_changed_filepaths = unstaged_files + staged_files
+    return [filepath for filepath in all_changed_filepaths]
+
+
+def _get_all_files_in_directory(dir_path, excluded_glob_patterns):
+    """Recursively collects all files in directory and
+    subdirectories of specified path.
+
+    Args:
+        dir_path: str. Path to the folder to be linted.
+        excluded_glob_patterns: set(str). Set of all glob patterns
+            to be excluded.
+
+    Returns:
+        a list of files in directory and subdirectories without excluded files.
+    """
+    files_in_directory = []
+    for _dir, _, files in os.walk(dir_path):
+        for file_name in files:
+            filepath = os.path.relpath(
+                os.path.join(_dir, file_name), os.getcwd())
+            if not any([fnmatch.fnmatch(filepath, gp) for gp in
+                        excluded_glob_patterns]):
+                files_in_directory.append(filepath)
+    return files_in_directory
+
+
+def _get_file_extensions(file_extensions_to_lint):
+    """This function is used to return the file extensions which need to be
+    linted and checked.
+
+    Args:
+        file_extensions_to_lint: list(str). The list of file extensions to be
+        linted checked.
+
+    Returns:
+        all_file_extensions_type: list(str). The list of all file extensions
+        to be linted and checked.
+    """
+    all_file_extensions_type = ['js', 'py', 'html', 'css', 'other']
+
+    if file_extensions_to_lint:
+        # Check if 'js' and 'ts' both are present in file_extensions_to_lint.
+        js_and_ts_is_present = 'js' in file_extensions_to_lint and (
+            'ts' in file_extensions_to_lint)
+
+        if js_and_ts_is_present:
+            python_utils.PRINT(
+                'Please use only one of "js" or "ts", as we do not have '
+                'separate linters for JS and TS files. If both these options '
+                'are used together, then the JS/TS linter will be run twice.')
+            python_utils.PRINT('Exiting...')
+            sys.exit(1)
+
```

Comment:
@Hudda Do we need to check whether all the passes params in `file_extensions_to_lint` are valid?

Reply:
@DubeySandeep validation is added [here](https://github.com/oppia/oppia/blob/9a1f6a25679dfbfa042aa1bba3364dec712a8af2/scripts/linters/pre_commit_linter.py#L93). This option in argparse only allows a limited subset of choices.